DESIGN.md

```diff
@@ -0,0 +1,52 @@
+# Overview
+Warnet is a software suite that allows its users to safely put ideas about Bitcoin to the test. It does this by using Kubernetes to simulate Bitcoin networks and providing user interaction. This document explains how Warnet does this by using design philosophies popular with Kubernetes such as Infrastructure as Code and a tendency towards Stateless Configuration and Externalized Configuration.
+
+# Project structure
+The Warnet code base has four main sections:
+
+1. resources - these items are available during runtime and relate to configuration (crucially, Kubernetes configuration)
+2. src/warnet - python source code lives here
+3. test - CI testing files live here
+4. docs - stores documentation available in the github repository
+
+## Overview of resources
+There are four main kinds of *resources*:
+
+1. Kubernetes configuration files - they are the backbone of Stateless Configuration; they are *yaml* files
+2. scenarios - these are python programs that users can load into Kubernetes to interact with the simulated Bitcoin network
+3. images - the logic for creating bitcoin nodes and also containers for running scenarios are found here; this includes Dockerfiles
+4. scripts and other configs - these are like "assets" or "one off" items which appear in Warnet.
```

Comment:
Could perhaps mention here (or somewhere else) that everything in `resources/` is packaged into the `warnet` python package. This way it's accessible by the `warnet` command once installed (via pip).

This is how we launch various process using their charts without needing to copy everything into every network directory, which should contain only a `network.yaml` and `node-defualts.yaml`.

Reply:
Ah I see now this is touched upon in the final section!